frontend/src/lib/monaco/codeEditorLogic.tsx

```diff
@@ -0,0 +1,129 @@
+                    const model = props.editor?.getModel()
+                    if (!model || !props.monaco || (props.language !== 'hogql' && props.language !== 'hog')) {
+                        return null
+                    }
+                    await breakpoint(300)
+                    const query = props.query
+                    const response = await performQuery<HogQLMetadata>(
+                        props.language === 'hogql'
+                            ? { kind: NodeKind.HogQLMetadata, select: query, filters: props.metadataFilters }
+                            : { kind: NodeKind.HogQLMetadata, program: query, filters: props.metadataFilters }
+                    )
+                    breakpoint()
+                    return [query, response]
+                },
+            },
+        ],
+        modelMarkers: [
+            [] as ModelMarker[],
+            {
+                reloadMetadataSuccess: ({ metadata }) => {
+                    const model = props.editor?.getModel()
+                    if (!model || !metadata) {
+                        return []
+                    }
+                    const markers: ModelMarker[] = []
+                    const [query, metadataResponse] = metadata
+
+                    function noticeToMarker(error: HogQLNotice, severity: MarkerSeverity): ModelMarker {
+                        const start = model!.getPositionAt(error.start ?? 0)
+                        const end = model!.getPositionAt(error.end ?? query.length)
+                        return {
+                            start: error.start ?? 0,
+                            startLineNumber: start.lineNumber,
+                            startColumn: start.column,
+                            end: error.end ?? query.length,
+                            endLineNumber: end.lineNumber,
+                            endColumn: end.column,
+                            message: error.message ?? 'Unknown error',
+                            severity: severity,
+                            hogQLFix: error.fix,
+                        }
+                    }
+
+                    for (const notice of metadataResponse?.errors ?? []) {
+                        markers.push(noticeToMarker(notice, 8 /* MarkerSeverity.Error */))
+                    }
+                    for (const notice of metadataResponse?.warnings ?? []) {
+                        markers.push(noticeToMarker(notice, 4 /* MarkerSeverity.Warning */))
+                    }
+                    for (const notice of metadataResponse?.notices ?? []) {
+                        markers.push(noticeToMarker(notice, 1 /* MarkerSeverity.Hint */))
+                    }
+
+                    props.monaco?.editor.setModelMarkers(model, 'hogql', markers)
+                    return markers
+                },
+            },
+        ],
+    })),
+    selectors({
+        isValidView: [(s) => [s.metadata], (metadata) => !!(metadata && metadata[1]?.isValidView)],
```

Comment:
metadata isn't updating now -> hogQLQueryEditorLogic connection to this field isn't updating either so this value is always false. Trying to debug right now but noting for visibility